web/saml.go

```diff
@@ -16,6 +16,8 @@ import (
 	"github.com/mattermost/mattermost-server/v6/utils"
 )
 
+const maxSAMLResponseSize = 2 * 1024 * 1024 // 2MB
```

Comment:
I'm not super confident on what this number should actually be, but the few points of comparison that I can find -- SAML SP implementations that set a limit on Response or Assertion lengths -- all set the limit to a few hundred kilobytes. Based on that, 2 MB should be plenty enough for interoperability.